### PR TITLE
Enhance error handling around Guerilla Email Client

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
## Category

- [ ] Bugfix
- [x] Enhancement
- [ ] New Feature
- [ ] Configuration Change
- [ ] Versioning Change
- [ ] Unit/Integration Test(s)
- [ ] Documentation

### CODE REVIEW

File: integration-tests/src/test/groovy/com/okta/authn/sdk/its/email/EmailClient.groovy [hunk 1]
```diff
@@ -93,12 +95,18 @@ class EmailClient {
 
     private Email createEmailAccount() {
 
+        Response response = get(GUERILLA_MAIL_BASE)
```
Comment (arvindkrishnakumar-okta) [inline integration-tests/src/test/groovy/com/okta/authn/sdk/its/email/EmailClient.groovy:98]: At the time of writing this `http://api.guerrillamail.com/ajax.php` is down and this causes the below tests to fail: 

`AuthenticationIT#forgotPasswordTest()`
`AuthenticationIT#unlockAccountTest()`

These tests depend on Guerilla Mail API to read recovery token sent (from Okta) to email address hosted by Guerilla Mail.

File: integration-tests/src/test/groovy/com/okta/authn/sdk/its/AuthenticationClientIT.groovy [hunk 2]
```diff
@@ -85,20 +85,21 @@ class AuthenticationClientIT extends AuthenticationTestSupport {
...
-        def factorVerifyRequest = authClient.instantiate(VerifyPassCodeFactorRequest)
-                                                            .setPassCode(totp.now())
-                                                            .setStateToken(response4.getStateToken())
-
-        def response5 = authClient.verifyFactor(factorId, factorVerifyRequest, ignoringStateHandler)
-        assertThat response5.getStatus(), is(AuthenticationStatus.SUCCESS)
-        assertThat response5.getSessionToken(), not(isEmptyString())
+        assertThat response4.getStatus(), is(AuthenticationStatus.SUCCESS)
+        assertThat response4.getSessionToken(), not(isEmptyString())
+//        assertThat response4.getSessionToken(), nullValue()
```
Comment (arvindkrishnakumar-okta) [inline integration-tests/src/test/groovy/com/okta/authn/sdk/its/AuthenticationClientIT.groovy:90]: The next step in MFA might no longer be required as the test results suggest. I am not entirely sure if that's the case going forward and hence commented the below lines.

File: integration-tests/src/test/groovy/com/okta/authn/sdk/its/AuthenticationClientIT.groovy [hunk 3]
```diff
@@ -85,20 +85,21 @@ class AuthenticationClientIT extends AuthenticationTestSupport {
...
+//        def factorVerifyRequest = authClient.instantiate(VerifyPassCodeFactorRequest)
+//                                                            .setPassCode(totp.now())
+//                                                            .setStateToken(response4.getStateToken())
+//
+//        def response5 = authClient.verifyFactor(factorId, factorVerifyRequest, ignoringStateHandler)
+//        assertThat response5.getStatus(), is(AuthenticationStatus.SUCCESS)
+//        assertThat response5.getSessionToken(), not(isEmptyString())
     }
 
     @Test(groups = "email")
```
Comment (arvindkrishnakumar-okta) [inline integration-tests/src/test/groovy/com/okta/authn/sdk/its/AuthenticationClientIT.groovy:105]: We might need to disable this test temporarily as this fails presently due to the outage of Guerilla Mail API.

**Review by JayNewstrom — APPROVED** [review]

**Review by bdemers — APPROVED** [review]
💯

**Pull request merged**